master/static/srv/task-signal-handling.sh

```diff
@@ -36,7 +36,7 @@ wait_and_handle_signals() {
         # When a signal is sent to the shell, it will interrupt waits, after all traps have run. To
         # discern if the wait unblocked because of a signal or process exit, we set "trapped_signal"
         # in traps and check it here.
-        if [ -z "${trapped_signal}" ]; then
+        if [ -z ${var+trapped_signal} ]; then
```

Comment:
This works (I manually tested it), but oddly doesn't seem to be documented for bash - https://www.gnu.org/software/bash/manual/bash.html#Shell-Parameter-Expansion

It is documented in the opengroup shell specification https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02